cocos/2d/CCNode.cpp

```diff
@@ -1364,9 +1367,19 @@ void Node::onEnter()
 #endif
     
     _isTransitionFinished = false;
-    
-    for( const auto &child: _children)
-        child->onEnter();
+
+    _childrenDirty = false;
+    for (ssize_t i = 0; i < _children.size(); i++)
+    {
+        auto child = _children.at(i);
+        if (!child->isRunning())
+            child->onEnter();
+        if (_childrenDirty)
+        {
+            i = -1;
```

Comment:
It will loop from begin again?
Some children's `onEnter` will be invoked more than once.